### PR TITLE
Refactor logical routines from `core.pyx`

### PR DESCRIPTION
**Pull request by asi1024**

### CODE REVIEW

**Comment by leofang** [comment]
Jenkins, test this please

File: cupy/core/_routines_logic.pxd [hunk 1]
```diff
@@ -3,3 +3,9 @@ from cupy.core.core cimport ndarray
 
 cdef ndarray _ndarray_all(ndarray self, axis, out, keepdims)
 cdef ndarray _ndarray_any(ndarray self, axis, out, keepdims)
+cdef ndarray _ndarray_greater(ndarray self, other)
+cdef ndarray _ndarray_greater_equal(ndarray self, other)
+cdef ndarray _ndarray_less(ndarray self, other)
+cdef ndarray _ndarray_less_equal(ndarray self, other)
+cdef ndarray _ndarray_equal(ndarray self, other)
+cdef ndarray _ndarray_not_equal(ndarray self, other)
```
Comment (leofang) [inline cupy/core/_routines_logic.pxd:11]: I don't think my question is specific to this PR, but I recently begin to wonder: why don't we inline these functions? They are module-level functions which are perfect target for inline, say,
```python
cdef inline ndarray _ndarray_not_equal(ndarray self, other)
```
Then they'll be inlined to the array methods. @asi1024 Do you know?

Comment (asi1024) [reply cupy/core/_routines_logic.pxd:11]: Sorry, I don't know..., but I think it's OK to let them be inlined.

**Comment by leofang** [comment]
CI was dead. Retry...

Jenkins, test this please

**Review by emcastillo — APPROVED** [review]
LGTM!

**Pull request merged**